src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let elem of this) {
+      if (elem === null || elem === undefined) {
+        elem = '';
+      }
+
+      if (separator === null) {
+        result += 'null' + elem;
+      } else {
+        result += separator + elem;
+      }
+    }
+
+    return result.slice(String(separator).length, result.length);
```

Comment:
instead of using slice, you can just add separator if element index not '0' (in this case you need to use 
"for (let i=0; i < this.length; i+=1 )" )

Reply:
thank you, I'll fix it